### PR TITLE
virt-test: fix bug in reboot function

### PR DESCRIPTION
**Pull request by xutian**
1. update reboot keywords pattern
2. don't close session until search reboot keywords timeout
3. catch exception when it raise in  match keywords

Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
ID: 1205993

**Comment by spcui** [comment]
Acked-by: @spcui

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -3578,14 +3578,14 @@ def __reboot(session):
...
-            else:
-                timeout = self.CLOSE_SESSION_TIMEOUT
-                ret = not session.is_responsive(timeout=timeout)
-            session.close()
-            return ret
+                patterns = [r".*[Rr]ebooting.*", r".*[Rr]estarting system.*",
+                            r".*[Mm]achine restart.*", r".*Linux version.*"]
+                try:
+                    return session.read_until_any_line_matches(patterns,
+                                                               timeout=60)
```
Comment (ypu) [inline virttest/qemu_vm.py]: timeout here may bigger than below code. So it may cause problem in our test.

**Comment by xutian** [comment]
@ypu  updated, please review. thanks

**Comment by FengYang** [comment]
It looks good to me.

Acked-by: FengYang

**Pull request merged**